experiment/contractNipopow.sol

```diff
@@ -189,11 +193,16 @@ contract Crosschain {
                 reversedSiblings[j] = siblings[ptr + j];
             ptr += branchLength;
 
+            // We only care to verify within our range
+            if (i < startValidateIndex || endValidateIndex < i) {
+                continue;
+            }
```

Comment:
This can be directly incorporated in the for loop

Reply:
We have to iterate through siblings and update the `ptr`, because siblings array is not aligned with the headers array.

Reply:
But this is ugly.
I can keep `validateInterlink` as it was before and have a different function to find the `siblings` offset and then call directly `validateMerkle`.